### DISCUSSION TITLE
Is it possible to run Whisper cpp with batch of wav files ?

### DISCUSSION THREAD

**Discussion by amitli1** [ggml-org/whisper.cpp#652] [General]
We can run Whisper cpp with one file.
Is it possible to run Whisper cpp with batch of wav files ?

**Comment by misutoneko** [comment]
Of course, it's quite handy with libfvad for example. Never used --help huh?
I think the context is lost between the files (not sure about that though - please correct me if that's wrong).

**Reply by amitli1** [reply]
I used help, but didn't saw it is possible.

What is the command to run batch of wav files ?

For example, when using batch with size 1 (single file), I'm using the following command:
`./main -m models/ggml-base.bin -l iw -bs 8 -f example1.wav`

It seems that `-f` take only 1 file.

**Comment by misutoneko** [comment]
Yes I see what you mean, it's not very clear. You can simply use multiple -f definitions, like so:

> ./main -m models/ggml-base.bin -l iw -bs 8 -f example1.wav -f example2.wav

**Reply by amitli1** [reply]
I tried and it seems that it run the files in sequential and not parallel (as batch works in deep learning).
Have you tried and got shorter running times ?

**Comment by misutoneko** [comment]
AFAIK while there is parallel processing (the -t switch) in whisper.cpp, it isn't implemented for multiple files. The same thing can be accomplished with an external setup that launches several instances of the main binary (xargs -P maybe?). Although you have to be careful with memory usage with that kind of setup.

In my own scripts I have a libfvad preprocess stage which generates a bunch of (generally very small) wav files, so parallel processing would be a good fit. The same thing could be done by whisper.cpp itself and it might give some more performance when the hw has lots of threads.
The running times are currently not an issue for me, so I haven't really pursued performance enhancements in this way.
